examples/platform/silabs/efr32/spi_multiplex.h

```diff
@@ -0,0 +1,38 @@
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include "FreeRTOS.h"
+#include "semphr.h"
+#include "sl_mx25_flash_shutdown_usart_config.h"
+#include "sl_spidrv_exp_config.h"
+#include "sl_wfx_host_api.h"
+
+#define LCD_BIT_RATE 1100000
+#define EXP_HDR_BIT_RATE 16000000
+#define SPI_FLASH_BIT_RATE 16000000
+
+typedef enum PERIPHERAL_TYPE
+{
+    EXP_HDR = 0,
+    LCD,
+    EXT_SPIFLASH,
+} peripheraltype_t;
+
```

Comment:
A suggestion that I find a bit cleaner instead of the 3 defines and the "if else if" sequence of `set_spi_baudrate`
```suggestion
const uint32_t kSpiBitRate[] = {
    [EXP_HDR] =  16000000,
    [LCD] = 1100000,
    [EXT_SPIFLASH] = 16000000,
};
```
then `set_spi_baudrate` could be simplified to a 1 liner or just replaced directly by
`spi_drv_reinit(kSpiBitRate[pr_type]);`

Reply:
Since these changes has already been merged to RC_1.0.1, these review comments would be in-corporated with another PR to maintain same code across all branches.  In order to track this, created JIRA ticket https://jira.silabs.com/browse/MATTER-1007.